prepare_data.py

```diff
@@ -30,54 +26,76 @@ def main():
+    parser.add_argument("--season", required=True, type=int, help="Current year (for naming purposes)")
     parser.add_argument(
         "--division", default="all", choices=["women", "mixed", "open", "all"], help="Division (women/mixed/open/all)"
     )
-    parser.add_argument("--season", required=True, type=int, help="Current year (for naming purposes)")
     parser.add_argument("--output", required=True, type=Path, help="Path to the folder to save the output CSVs")
     args = parser.parse_args()
 
+
     if args.division == "all":
         divisions = ["mixed", "open", "women"]  # Prepare data for all divisions at once
     else:
         divisions = [args.division]
 
-    setup_logger(args.output / f"prepare_data-{args.season}-{args.division}.log")
+    prepare_data(args.input, args.season, divisions, args.output)
+
+def prepare_data(input_path, season, divisions, output_path):
+    """
+    Take data from all the CSV files in the input folder and join them to create a big Game Table with the clean data;
+    export some preliminary summary statistics (no rankings are calculated here).
+
+    Prerequisites:
+    * Prepare a folder with the tournament result data - CSV files with columns Tournament, Date, Team_1, Team_2,
+      Score_1, Score_2, and Division
+    * Add to the same folder a TXT file per division specifying the teams in the EUF system; multiple aliases can be
+      defined for each team in the same row, separated with commas (filename should be teams-<division>.txt)
+    * Add to the same folder a TXT file with the pairs <team>, <tournament>; specifying that the given team has met the
+      EUF roster requirements for the particular tournament (filename should be teams_at_tournaments-<division>.txt)
+
+    Procedure:
+    * Read the tournament results CSVs and take only the games for the given division
+    * Read the list of EUF teams; replace aliases where applicable
+    * Read teams at tournaments list; add suffix to all teams without EUF-sanctioned roster for the given tournament
+    * Process the Game Table (check for invalid rows)
+    * Calculate basic statistics for the season (without any rankings)
+    * Save the output CSVs
+
+    :input_path: path to the folder with all necessary files
+    :season: current year
+    :divisions: list of divisions for which the ranking should be calculated
+    :output_path: path to the folder in which to save the output CSVs
+    """
+    os.makedirs(output_path, exist_ok=True)
+    setup_logger(output_path / f"prepare_data-{season}-{'_'.join(divisions)}.log")
     logger = logging.getLogger("ranking.data_preparation")
 
     for division in divisions:
-        logger.info(f"Preparing data for season {args.season}, {division} division.")
-        df_teams = pd.read_csv(args.input / f"teams-{division}.csv")
+        logger.info(f"Preparing data for season {season}, {division} division.")
```

Comment:
small nit pick, but imho you should generally take advantage of lazy logging in python
for reference: https://medium.com/flowe-ita/logging-should-be-lazy-bc6ac9816906

Reply:
I personally prefer cleaner and better-looking string formatting to some negligible time improvement.

Reply:
Regardless of whichever route we think is best, this pull request is just changing the line so as not to introduce a bug. How to deal with logging should not be part of this pull request in my mind